crates/router/src/core/payments/routing.rs

```diff
@@ -1322,22 +1322,28 @@ pub async fn perform_success_based_routing(
             .ok_or(errors::RoutingError::SuccessRateClientInitializationError)
             .attach_printable("success_rate gRPC client not found")?;
 
-        let success_based_routing_configs = routing::helpers::fetch_success_based_routing_configs(
-            state,
-            business_profile,
-            success_based_algo_ref
-                .algorithm_id_with_timestamp
-                .algorithm_id
-                .ok_or(errors::RoutingError::GenericNotFoundError {
-                    field: "success_based_routing_algorithm_id".to_string(),
-                })
-                .attach_printable(
-                    "success_based_routing_algorithm_id not found in business_profile",
-                )?,
-        )
-        .await
-        .change_context(errors::RoutingError::SuccessBasedRoutingConfigError)
-        .attach_printable("unable to fetch success_rate based dynamic routing configs")?;
+        let mut success_based_routing_configs =
+            routing::helpers::fetch_success_based_routing_configs(
+                state,
+                business_profile,
+                success_based_algo_ref
+                    .algorithm_id_with_timestamp
+                    .algorithm_id
+                    .ok_or(errors::RoutingError::GenericNotFoundError {
+                        field: "success_based_routing_algorithm_id".to_string(),
+                    })
+                    .attach_printable(
+                        "success_based_routing_algorithm_id not found in business_profile",
+                    )?,
+            )
+            .await
+            .change_context(errors::RoutingError::SuccessBasedRoutingConfigError)
+            .attach_printable("unable to fetch success_rate based dynamic routing configs")?;
+
+        success_based_routing_configs.config.as_mut().map(|config| {
+            config.specificity_level =
+                Some(api_models::routing::SuccessRateSpecificityLevel::Merchant)
+        });
```

Comment:
should we manually update this level in code or should we make use of `success_based_routing_update_configs` endpoint to update this in config